src/common/makeOutcomeFailDependingOnOutput.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import {ISpawnResult} from "./node/childProcess";
+
+export type PatternToFailure = { [pattern: string]: string };
+
+/* This class will transform a spawn process to only succed if all defined success patterns
+   are found on stdout, and none of the failure patterns were found on stderr */
+export class MakeOutcomeFailDependingOnOutput {
+    private generatePatternsForSuccess: () => Q.Promise<string[]>;
+    private generatePatternToFailure: () => Q.Promise<PatternToFailure>;
+
+    private output = "";
+    private errors = "";
+
+    constructor(generatePatternsForSuccess: () => Q.Promise<string[]>, generatePatternToFailure: () => Q.Promise<PatternToFailure>) {
+        this.generatePatternsForSuccess = generatePatternsForSuccess;
+        this.generatePatternToFailure = generatePatternToFailure;
+    }
+
+    public process(spawnResult: ISpawnResult): Q.Promise<void> {
+        // Store all output
+        this.store(spawnResult.stdout, new_content => this.output += new_content);
```

Comment:
It can, and there is no limit currently. Given that this is intended as a temporary measure until react-native returns proper exit codes, I'd leave it as it though. What do you think?